### PR TITLE
sumologicexporter: use Sumo Logic Extension for authentication and to obtain endpoint

### LINKED ISSUE #32375: Sample config from sumologic extension doesn't work
### Component(s)

extension/sumologic

## Description

I try tu run collector with this config: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/sumologicextension#example-config. It didn't start correctly.

## Steps to Reproduce

Try tu run collector with this config: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/sumologicextension#example-config
```
extensions:
  sumologic:
    installation_token: <token>
    collector_name: my_collector
    time_zone: Europe/Warsaw

receivers:
  hostmetrics:
    collection_interval: 30s
    scrapers:
      load:

processors:

exporters:
  sumologic:
    auth:
      authenticator: sumologic # Specify the name of the authenticator extension

service:
  extensions: [sumologic]
  pipelines:
    metrics:
      receivers: [hostmetrics]
      processors: []
      exporters: [sumologic]
```

## Expected Result

Collector should start

## Actual Result
Error:
```
Error: invalid configuration: exporters::sumologic: endpoint is not set
2024/04/15 11:27:02 collector server run finished with error: invalid configuration: exporters::sumologic: endpoint is not set
```

### Collector version

latest

## Environment
OS: darwin/amd64
Compiler(if manually compiled): go version go1.21.7

### OpenTelemetry Collector configuration

```yaml
extensions:
  sumologic:
    installation_token: <token>
    collector_name: my_collector
    time_zone: Europe/Warsaw

receivers:
  hostmetrics:
    collection_interval: 30s
    scrapers:
      load:

processors:

exporters:
  sumologic:
    auth:
      authenticator: sumologic # Specify the name of the authenticator extension

service:
  extensions: [sumologic]
  pipelines:
    metrics:
      receivers: [hostmetrics]
      processors: []
      exporters: [sumologic]
```

### Log output

```shell
Error: invalid configuration: exporters::sumologic: endpoint is not set
2024/04/15 11:27:02 collector server run finished with error: invalid configuration: exporters::sumologic: endpoint is not set
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by sumo-drosiek**
**Description:**

Use Sumo Logic Extension for authentication and to obtain the endpoint. This is part of #31479 to make #32315 smaller, therefore it may have some unused code (like `getSignalURL`). Also OTLP format is not supported yet for neither of the formats

It also partially fixes #32375 (partially because as long as there is no a otlp support, data won't be pushed to Sumo Logic)

**Link to tracking Issue:** #31479

**Testing:**

- unit tests
- manual tests

**Documentation:**

- ensures that Sumo Logic Extension documentation is valid

### CODE REVIEW

**Review by swiatekm — APPROVED** [review]
LGTM, one minor nitpick.

File: exporter/sumologicexporter/config.go [hunk 1]
```diff
@@ -60,11 +63,14 @@ type Config struct {
 	Client string `mapstructure:"client"`
 }
 
-// createDefaultClientConfig returns default http client settings
+// CreateDefaultClientConfig returns default http client settings
```
Comment (sumo-drosiek) [inline exporter/sumologicexporter/config.go:66]:
```suggestion
// createDefaultClientConfig returns default http client settings
```

File: exporter/sumologicexporter/factory.go [hunk 2]
```diff
@@ -17,6 +17,14 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/sumologicexporter/internal/metadata"
 )
 
+const (
+	// The value of "type" key in configuration.
+	typeStr        = "sumologic"
+	stabilityLevel = component.StabilityLevelBeta
```
Comment (astencel-sumo) [inline exporter/sumologicexporter/factory.go:23]: These consts should be removed, you should use the generated metadata code from `internal/metadata/generated_status.go`: `metadata.Type`, `metadata.*Stability`.

Comment (sumo-drosiek) [reply exporter/sumologicexporter/factory.go]: Done

**Review by andrzej-stencel — APPROVED** [review]

**Pull request merged**